lua/gruvbox.lua

```diff
@@ -71,60 +71,59 @@ Gruvbox.config = {
 -- main gruvbox color palette
 ---@class GruvboxPalette
 Gruvbox.palette = {
-  dark0_hard = "#1d2021",
-  dark0 = "#282828",
-  dark0_soft = "#32302f",
-  dark1 = "#3c3836",
-  dark2 = "#504945",
-  dark3 = "#665c54",
-  dark4 = "#7c6f64",
-  light0_hard = "#f9f5d7",
-  light0 = "#fbf1c7",
-  light0_soft = "#f2e5bc",
-  light1 = "#ebdbb2",
-  light2 = "#d5c4a1",
-  light3 = "#bdae93",
-  light4 = "#a89984",
-  bright_red = "#fb4934",
-  bright_green = "#b8bb26",
-  bright_yellow = "#fabd2f",
-  bright_blue = "#83a598",
-  bright_purple = "#d3869b",
-  bright_aqua = "#8ec07c",
-  bright_orange = "#fe8019",
-  neutral_red = "#cc241d",
-  neutral_green = "#98971a",
-  neutral_yellow = "#d79921",
-  neutral_blue = "#458588",
-  neutral_purple = "#b16286",
-  neutral_aqua = "#689d6a",
-  neutral_orange = "#d65d0e",
-  faded_red = "#9d0006",
-  faded_green = "#79740e",
-  faded_yellow = "#b57614",
-  faded_blue = "#076678",
-  faded_purple = "#8f3f71",
-  faded_aqua = "#427b58",
-  faded_orange = "#af3a03",
-  dark_red_hard = "#792329",
-  dark_red = "#722529",
-  dark_red_soft = "#7b2c2f",
-  light_red_hard = "#fc9690",
-  light_red = "#fc9487",
-  light_red_soft = "#f78b7f",
-  dark_green_hard = "#5a633a",
-  dark_green = "#62693e",
-  dark_green_soft = "#686d43",
-  light_green_hard = "#d3d6a5",
-  light_green = "#d5d39b",
-  light_green_soft = "#cecb94",
-  dark_aqua_hard = "#3e4934",
-  dark_aqua = "#49503b",
-  dark_aqua_soft = "#525742",
-  light_aqua_hard = "#e6e9c1",
-  light_aqua = "#e8e5b5",
-  light_aqua_soft = "#e1dbac",
-  gray = "#928374",
+ dark0 = { ["gui"] = "#282828", ["cterm"] = 235 },
```

Comment:
isn't it easier to read this way? --
```suggestion
 dark0 = { gui = "#282828", cterm = 235 },
```

Reply:
1. This patch removes `dark0_hard` for whatever reason.

2. why not structure this as:
```
Gruvbox.palette = {
  gui = {
    dark0_hard = "#1d2021",
    dark0 = "#282828",
    ...
  },
  cterm = {
    dark0_hard = 235,
    dark0 = 235,
    ...
  }
}
```

Reply:
Thanks for reviewing. 

0. Readibility without brackets: I fully agree. Sorry, it is the first time I'm coding lua and I thought for some reason that the brackets ought to be there. Anyhow, corrected now in the most recent commit. 
1. removal of `dark0_hard` was unintentional, thanks for spotting that. Corrected in the most recent commit. 
2. There were three reasons for it in my mind. First, it's supposed to be the same color, so it feels natural to have one key that then is used to access the same color for two different ways to configure the color. Second, the names are used eg. in the color groups. I felt it would require many more changes in the code and would result in a lot of overhead code. Now it the main logic is the same and at the end, for all highlights, a conversion expands into gui or cterm. The third reason is maintainability. It is much better to have the two values close to each other in the code. If you change the gui value, you see you need to adapt the cterm value as well. What would be your argument for changing it that way?

Reply:
The advantage of the proposed structure is a vast simplification of the patch: the entire `expand_colors_in_highlights()` becomes unnecessarily as the correct palette can be returned from `get_colors()` depending on the value of `vim.o.termguicolors`, everything else (excluding your patch in L1242) can remain the same as before.

Reply:
@jomade Here's what I could come up with -- https://github.com/lxv/gruvbox.nvim/commit/eae39ed81238a6c65a562ad9617404bec5ff68a6 -- I'm not entirely sure if it's that much better than your version. It's disappointing that unlike `vim_get_hl_by_name()` which allows users to specify GUI/TUI flag in the 2nd arg, `nvim_set_hl()` doesn't support that (ref: https://github.com/neovim/neovim/blob/master/test/functional/api/highlight_spec.lua)

Reply:
NVM, my version breaks runtime changes `termguicolors` <-> `notermguicolors` because it only has access to either `cterm` or `gui` color value at a time. I'm sorry about the noise.

Reply:
No worries.
Btw, I know see that I probably can make the expansion function more compact. Let's see if I can get that done on one of the next evenings. Sth along the lines (not tested): 
```
local function expand_highlight_prop(hl, prop)
   if hl.[prop] == nil then return end
   if hl.[prob] == "NONE" then
     if prop != "sp" then 
          hl.["cterm"..prop] = "NONE"
     end
     hl.[prop] = "NONE"
   else 
     hl.["cterm"..prop] = hl.[prop].cterm
     hl.[prop] = hl.[prop].gui
   end
end
``` 
And then call the function for `"fg"`, `"bg"` and `"sp"`.

Reply:
See my recent upgrade. I have compacted the code as outlined here (minus the bugs in the displayed code...)